pyquil/operator_estimation.py

```diff
@@ -0,0 +1,428 @@
+import itertools
+import json
+import logging
+from json import JSONEncoder
+from math import pi
+from typing import List, Union
+
+import networkx as nx
+import numpy as np
+from dataclasses import dataclass
+from networkx.algorithms.approximation.clique import clique_removal
+
+from pyquil import Program
+from pyquil.api import QuantumComputer
+from pyquil.gates import *
+from pyquil.paulis import PauliTerm
+
+log = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class Experiment:
```

Comment:
Yes that's true. Maybe `Experiment` is the wrong name for this class.

`ExperimentSuite` has the program and a list of `Experiment`s, each of which is a tuple `(state_prep, measure_basis)`. This is because typically we want to measure one program (ansatz) in many different ways with many different state preps.